<div class='content'>
<style>
.limited200,
.height200{max-height:200px; overflow-y: scroll;}
</style>
	<h2>项目描述</h2>
	<pre class=saying>假如一个人的精力是10，工作用了7，游戏用了2，最后只要1的精力去学习，而学习是对于你来说很重要的事情，那你能做好吗？
学生时代不乏听说谁谁谁，游戏打得好，学习也好，对不起不存在这样的人，就算真的有，其实他可以把学习做的更好！</pre>
	<pre class=saying>你从80楼往下看，全是美景，但你从2楼往下看，全是垃圾；人若没有高度，看到的全是问题，人若没有格局，看到的全是鸡毛蒜皮。	</pre>

	<p>scSeq分析主要包括表达矩阵(matrix)的获取，及后面步骤。</p>
	<p>主要流程包括：分离单细胞测序数据到单文件、质控与前处理、比对及质控、定量及表达矩阵，细胞质控。分群分析、差异表达分析等。</p>	
	
	<p class=red>高级分析: doublet 检测，RNA velocity分析, 细胞间通信。</p>
	<p>扫盲: 
		<a href="https://carpentries-incubator.github.io/scrna-seq-analysis/aio/index.html">大细胞扫盲</a> | 
	
	</p>





	<h2>好用的搜索</h2>
	<p>https://www.qinai.ml/</p>
	<p>https://fsoufsou.com/</p>
	<p>https://xgoogle.xyz/</p>
	<p>https://search.library.edu.eu.org/</p>
	<p>https://g.luciaz.me/ 答案: 最大板块"心灵之约"; 本科"水朝夕"; 学院"csxy@123"</p>
	<p>https://shitu.paodekuaiweixinqun.com</p>
	<p>https://search.iwiki.uk/ 无法访问</p>
<pre>
https://ceres.shuu.cf/
</pre>




	<h2>分析相关</h2>
	<p>scRNAseq分析教程: 
		<a target="_blank" class='red' href='https://scrnaseq-course.cog.sanger.ac.uk/website/index.html'>hemberg-lab《Analysis of single cell RNA-seq data》</a>
		<a target="_blank" class='button blue' href='https://github.com/hemberg-lab/scRNA.seq.course'>github</a>
		| 
		<a target="_blank" class='blue' href='https://ucdavis-bioinformatics-training.github.io/'>2021 August Single Cell RNA Seq Analysis</a>
		<a target="_blank" class='button blue' href='https://ucdavis-bioinformatics-training.github.io/2017_2018-single-cell-RNA-sequencing-Workshop-UCD_UCB_UCSF/day3/scRNA_Workshop-PART5.html'>2017/2018老教程</a>
		<a target="_blank" class='button red' href='https://ucdavis-bioinformatics-training.github.io/2017_2018-single-cell-RNA-sequencing-Workshop-UCD_UCB_UCSF/'>at UCD,UCB,UCSF</a>
		| 
		<a target="_blank" class='button blue' href='https://broadinstitute.github.io/2020_scWorkshop/' title="ANALYSIS OF SINGLE CELL RNA-SEQ DATA">broad</a>
	</p>






	<p>处理单细胞数据的众多R包：
		<a target="_blank" class='button yellow' href='https://osca.bioconductor.org/'>Orchestrating Single-Cell Analysis with Bioconductor</a>
	</p>


	<p>比较好的博客：
		<a target="_blank" class='button orange' href='https://www.jianshu.com/u/d7b77c171c15'>刘小泽 简书</a> 
		|
		<a target="_blank" class='button blue' href='https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&mid=2247496154&idx=3&sn=d3cfaa4a5b18235e0192619f64641635' title="">单细胞初级8讲和高级分析8讲</a>
	</p>
	
	<p> single cell dataset: 
		<a target="_blank" class='button blue' href='https://pubmed.ncbi.nlm.nih.gov/28416714/' title="A Repository of Immune-Related Single-Cell RNA-Sequencing Datasets">JingleBells(sc免疫数据集)</a>
	</p>
		
	<p>10X genomics资料: 
		<a target="_blank" href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">10x 官方数据集</a> | 
		<a target="_blank" href="https://www.10xgenomics.com/resources/publications">10x 文章</a> | 
		<a target="_blank" href="https://www.10xgenomics.com/blog/single-cell-rna-seq-an-introductory-overview-and-tools-for-getting-started">单细胞技术扫盲</a> | 
		<a target="_blank" href="https://data.humancellatlas.org/analyze/portals/cellxgene">cellxgene</a>(基于py和js的单细胞可视化工具) | 
	</p>







	<h2>C1, Fluidigm integrated fluidic circuit (IFC): C1 mRNA Seq HT IFC</h2>
	<p>
		<a target="_blank" href="http://cn.fluidigm.com/applications/single-cell-analysis">C1 single cell</a> | 
		<a target="_blank" href="https://www.fluidigm.com/products/c1-system">c1-system</a> | 
		<a target="_blank" href="https://www.fluidigm.com/publications/c1">paper</a> | 
		<a target="_blank" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4812709/">C1 鼠胰岛scRNA</a> | 
	</p>
	<p>C1, The first automated solution for single-cell genomics, now capable of even more. </p>
	<p>Reflected in over 100 published studies, the breadth of C1 applications empowers users to survey cell diversity, identify rare cell types and characterize cellular functions, all on the same single-cell biology platform. Investigate tissue heterogeneity with transcript end counting. Use T cell receptor sequencing and other full-length mRNA-seq protocols to recognize isoform- and receptor-specific expression. Examine gene-specific and microRNA expression. Perform epigenetic analysis and identify genomic variants. The C1 system delivers power and flexibility to support you all along your path to discovery.</p>
<pre class=height200>
cell lines:

Single-cell analysis of lung adenocarcinoma cell lines reveals diverse expression patterns of individual cells invoked by a molecular target drug treatment
Suzuki, A., Matsushima, K., Makinoshima, H. et al. Genome Biology (2015): 66

Single-Cell Transcriptomics Identifies Intra-Tumor Heterogeneity in Human Myeloma Cell Lines
Mitra, A.K., Stessman, H., Linden, M.A., Van Ness, B. Blood (2014): 3385

# 细胞间异质性
Aging increases cell-to-cell transcriptional variability upon immune stimulation
Martinez-Jimenez, C.P, Eling, N., Chen, H.C. et al. Science (2017): 1,433–6

# 干细胞
Single-Cell Transcriptomic Analysis Defines Heterogeneity and Transcriptional Dynamics in the Adult Neural Stem Cell Lineage
Dulken, B.W., Leeman, D.S., Boutet, S.C. et al. Cell Reports (2017): 777–90
# 干细胞
A Panel of Embryonic Stem Cell Lines Reveals the Variety and Dynamic of Pluripotent States in Rabbits
Osteil, P., Moulin, A., Santamaria, C. et al. Stem Cell Reports (2016): 383–98
</pre>

	<h2>10x scRNAseq</h2>
	<p>
		<a target="_blank" href="https://support.10xgenomics.com/single-cell-gene-expression">support.10xgenomics</a> | 
		<a target="_blank" href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">datasets</a> | 
	</p>
	
	<h3>V2</h3>
	<img src="/data/scSeq/images/sc_3_v2.png" />
	<pre>i. Partial Illumina Read 1 sequence (22 nucleotides (nt))
ii. 16 nt 10x™ Barcode
iii. 10 nt Unique Molecular Identifier (UMI)
iv. 30 nt Poly(dT) primer sequence</pre>
	<img src="/data/scSeq/images/10x_scRNAseq_v2.png" />
	<p>Fig. 2. Schematic of a fragment from a final Chromium™ Single Cell 3’ v2 library. *Can be adjusted</p>
	
	
	
	
	<h3>V3</h3>
	<p>v3.1 Chemistry: The <b>16 bp 10x Barcode and 12 bp UMI</b> are encoded in Read 1, while Read 2 is used to sequence the cDNA fragment.</p>
	<img src="/data/scSeq/images/10x_scRNAseq_v3.png" />

	<p>**Single Cell 3’ Gene Expression v2 libraries may be sequenced using the configuration for Single Cell 3’ Gene
Expression v3 libraries. However, v3 libraries should not be sequenced using the v2 configuration as 28 Read 1
cycles are required to capture the 16 nt 10x Barcode and the 12 nt UMI sequences.</p>







	<h2>bioconductor R包</h2>
	<p>
		<ul>
			<li>S4 Classes for Single Cell Data. <a target="_blank" class='button yellow' href='http://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html'>SingleCellExperiment 包</a> </li>

			<li>Seurat教程: <a target="_blank" class='button red' href='https://satijalab.org/seurat/vignettes.html'>Seurat</a> </li>
			<li>A tool for unsupervised clustering and analysis of single cell RNA-Seq data. <a target="_blank" class='button yellow' href='http://www.bioconductor.org/packages/release/bioc/html/SC3.html'>SC3 包</a> </li>
			<li>Clustering, differential expression, and trajectory analysis for single- cell RNA-Seq. <a target="_blank" class='button yellow' href='http://www.bioconductor.org/packages/release/bioc/html/monocle.html'>monocle 包</a> </li>
		</ul>
	</p>




<table>
  <thead>
    <tr>
      <th style="text-align:center">用法</th>
      <th style="text-align:center">Seurat 2.x</th>
      <th style="text-align:center">Seurat 3.x</th>
      <th>Scater</th>
      <th>Monocle2.x</th>
      <th>Monocle3.x</th></tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align:center">创建R包要求的对象</td>
      <td style="text-align:center">CreateSeuratObject()</td>
      <td style="text-align:center">函数不变，参数取消了raw.data，min.genes更改为min.features</td>
      <td>SingleCellExperiment()</td>
      <td>newCellDataSet()，其中的phenoData、featureData参数都是用new()建立的AnnotatedDataFrame对象</td>
      <td>new_cell_data_set()，其中的cell_metadata、gene_metadata参数都是数据框</td></tr>
    <tr>
      <td style="text-align:center">添加注释信息</td>
      <td style="text-align:center">AddMetaData()</td>
      <td style="text-align:center">AddMetaData()或者直接通过object$meta_name</td>
      <td>可以直接使用sce$meta_name</td>
      <td>addCellType()添加细胞类型</td>
      <td>可以用基础R函数</td></tr>
    <tr>
      <td style="text-align:center">QC and selecting cell</td>
      <td style="text-align:center">sce@raw.data</td>
      <td style="text-align:center">GetAssayData()</td>
      <td>calculateQCMetrics(),其中的feature_controls参数可以指定过滤指标，然后有一系列的可视化函数。过滤用filter()或isOutlier()</td>
      <td>用基础R函数进行初步过滤，还可以用detectGenes()函数加上subset()过滤</td>
      <td>用基础R函数进行初步过滤</td></tr>
    <tr>
      <td style="text-align:center">表达量的标准化或者归一化</td>
      <td style="text-align:center">NormalizeData()，归一化后检测用sce@data</td>
      <td style="text-align:center">NormalizeData()，归一化后检测用sce[['RNA']]</td>
      <td>计算CPM：calculateCPM()、归一化：normalize()</td>
      <td>estimateSizeFactors()还有estimateDispersions</td>
      <td>preprocess_cds()</td></tr>
    <tr>
      <td style="text-align:center">寻找重要的基因</td>
      <td style="text-align:center">FindVariableGenes()</td>
      <td style="text-align:center">FindVariableFeatures()，其中算法有变动</td>
      <td>没有专门函数</td>
      <td>differentialGeneTest()函数</td>
      <td>版本3和版本2的差异分析可以说是完全不同，版本3取代了2中的differentialGeneTest() and BEAM()。它利用fit_models()或graph_test()</td></tr>
    <tr>
      <td style="text-align:center">去除干扰因素</td>
      <td style="text-align:center">ScaleData()，结果存储在sce@scale.data中</td>
      <td style="text-align:center">ScaleData()，结果存储在sce[["RNA"]]@scale.data中</td>
      <td>limma的removeBatchEffect()、scran的mnnCorrect()</td>
      <td>去除干扰因素的功能被包装在降维函数中</td>
      <td>preprocess_cds()中指定参数residual_model_formula_str</td></tr>
    <tr>
      <td style="text-align:center">降维</td>
      <td style="text-align:center">PCA：RunPCA()，参数pc.genes，结果存储在sce@dr$pca@gene.loadings tSNE：RunTSNE()</td>
      <td style="text-align:center">PCA：RunPCA()，参数features，结果存储在sce@reductions$pca@feature.loadings tSNE：RunTSNE()</td>
      <td>PCA：runPCA()，结果在reducedDims中； tSNE：runTSNE()</td>
      <td>reduceDimension函数，可以选择多种参数</td>
      <td>reduce_dimension()，算法包括UMAP", "tSNE", "PCA" and "LSI"</td></tr>
    <tr>
      <td style="text-align:center">降维后可视化</td>
      <td style="text-align:center">VizPCA和PCElbowPlot;PCAPlot或者TSNEPlot</td>
      <td style="text-align:center">VizDimLoadings()、DimPlot()、DimHeatmap()、ElbowPlot()</td>
      <td>plotReducedDim()、plotPCA()</td>
      <td>plot_cell_clusters()</td>
      <td>plot_cells()</td></tr>
    <tr>
      <td style="text-align:center">细胞聚类</td>
      <td style="text-align:center">FindClusters()</td>
      <td style="text-align:center">FindNeighbors() + FindClusters()</td>
      <td>没有包装聚类函数，可以辅助其它R包，或者R基础函数</td>
      <td>clusterCells()</td>
      <td>cluster_cells()，依赖一个Python模块louvain</td></tr>
    <tr>
      <td style="text-align:center">找marker基因</td>
      <td style="text-align:center">FindMarkers()或FindAllMarkers()</td>
      <td style="text-align:center">FindMarkers()或FindAllMarkers()，VlnPlot()、FeaturePlot()可视化</td>
      <td>借助SC3包</td>
      <td>newCellTypeHierarchy()、 classifyCells()</td>
      <td>top_markers()</td></tr>
    <tr>
      <td style="text-align:center">绘图相关</td>
      <td style="text-align:center">基因相关性绘图：GenePlot()；细胞相关性绘图：CellPlot()，选择细胞用sce@cell.names</td>
      <td style="text-align:center">基因相关性绘图：FeatureScatter()；细胞相关性绘图：CellScatter()，选择细胞用colnames(sce)</td>
      <td>基因相关性绘图：绘制基因表达相关plotExpression()；检测高表达基因plotHighestExprs()、表达频率plotExprsFreqVsMean()、细胞质控plotColData()、表达量累计贡献plotScater()</td>
      <td>plot_cell_trajectory()、plot_genes_in_pseudotime()、plot_genes_jitter()、plot_pseudotime_heatmap()、plot_genes_branched_heatmap()、plot_genes_branched_pseudotime()</td>
      <td>plot_pc_variance_explained()、对每组的marker基因可视化： plot_genes_by_group()、3D发育轨迹plot_cells_3d()、画小提琴图：plot_genes_violin()</td></tr>
  </tbody>
</table>





	
<pre>
AUCell: Analysis of 'gene set' activity in single-cell RNA-seq data (e.g. identify cells with specific gene signatures)
LineagePulse: Differential expression analysis and model fitting for single-cell RNA-seq data
motifbreakR: A Package For Predicting The Disruptiveness Of Single Nucleotide Polymorphisms On Transcription Factor Binding Sites


scRNA分析使用的包：Ideal” scRNAseq pipeline (as of Oct 2017)
https://hemberg-lab.github.io/scRNA.seq.course/ideal-scrnaseq-pipeline-as-of-oct-2017.html


todo:
主题:	简介
Salmon定量实战	不基于比对直接定量基因和转录本的表达
差异基因分析: DEseq2	样本聚类热图、PCA、火山图、差异热图
GO、KEGG富集分析和可视化	R包，Cytoscape，泡泡图，网络图
GSEA富集分析，enrichMap	GSEA时间序列或相关性富集分析

STAR比对拼装差异剪接	和差异基因分析
WGCNA基因加权共表达:	共表达网络、Hub基因和性状关联热图
Cytoscape绘制 PPI互作: KEGG调控通路网络图+基因表达
常见生信图表解读:	Illustrator进行CNS修图和排版

单细胞转录组特点介绍:	不同技术比较、适用性和注意事项
单细胞数据分析和预处理:	Cellranger分析，细胞和基因筛选
单细胞分型:	Seurat, Scater, PCA, TSNE, SC3聚类
单细胞发育演化分析:	Pseudotime, Monocle，细胞周期鉴定
单细胞Marker基因鉴定:	Scran, 差异分析，功能分析



北大开的单细胞大会
视频 https://space.bilibili.com/16813275/#/
对应的笔记： https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247487815&idx=1&sn=8df778e3704563a085a89f5cecaa2dc5&scene=21#wechat_redirect



Date: 20190912
</pre>





</div>
